src/constructor/logger.py

```diff
@@ -0,0 +1,171 @@
+import glob
+import datetime
+from pathlib import Path
+from typing import Any, Dict, Optional, Union, Tuple
+from pytorch_lightning.loggers.base import LightningLoggerBase
+from pytorch_lightning.loggers import mlflow
+from pytorch_lightning.loggers.tensorboard import TensorBoardLogger
+from pytorch_lightning.loggers.wandb import WandbLogger
+from pytorch_lightning.loggers.csv_logs import CSVLogger
+from pytorch_lightning.loggers.neptune import NeptuneLogger
+from pytorch_lightning.loggers.mlflow import rank_zero_only
+
+
+def create_outputs_path(log_dir: str, experiment_name: str, create_datetime_log_subdir: bool
+                        ) -> Tuple[str, str]:
+    """Create directory for saving checkpoints and logging metrics.
+
+    Args:
+        log_dir: Base path.
+        experiment_name: Sub directory for log_dir.
+        create_datetime_log_subdir: If create datetime sub directory for log_dir / experiment_name directory. This
+            datetime would be some kind a version of experiment.
+
+    Returns:
+        experiment_subdir: A string of created datetime sub directory for log_dir / experiment_name.
+        full_outputs_path: Directory path to save checkpoints and logging metrics.
+    """
+    if create_datetime_log_subdir:
+        experiment_subdir = datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S")
+    else:
+        experiment_subdir = ''
+
+    log_dir = Path(log_dir)
+    full_outputs_path = log_dir / experiment_name / experiment_subdir
+    full_outputs_path.mkdir(exist_ok=True, parents=True)
+
+    return experiment_subdir, full_outputs_path
+
+
+class MLFlowLogger(mlflow.MLFlowLogger):
```

Comment:
Rename the class to `MLFlowLoggerX` and allow a user to choose between both: our version and Lighting's one